File: crabzilla-vertx/src/main/java/io/github/crabzilla/vertx/projection/ProjectionData.java
package io.github.crabzilla.vertx.projection;

import io.github.crabzilla.core.DomainEvent;
import lombok.Value;

import java.io.Serializable;
import java.util.List;
import java.util.UUID;

@Value
public class ProjectionData implements Serializable {

  private UUID uowId;
  private Long uowSequence;
  private String targetId;
  private List<DomainEvent> events;

}

File: crabzilla-vertx/src/main/java/io/github/crabzilla/vertx/projection/ProjectionData.kt
package io.github.crabzilla.vertx.projection

import io.github.crabzilla.core.DomainEvent
import java.util.*

data class ProjectionData(val uowId: UUID, val uowSequence: Long?,
                          val targetId: String, val events: List<DomainEvent>)
